### PR TITLE
docs: describe mise behavior when mise version is lower than min_version

### PR DESCRIPTION
**Pull request by erickguan**
Thanks for fixing my previous PR. This PR addresses [@glasser's comment](https://github.com/jdx/mise/pull/2989/files#r1837006850).

### CODE REVIEW

**Comment by glasser** [comment]
Thanks. Also it's not clear to me what section this key goes in — the other `###` headings are section names?

**Comment by glasser** [comment]
Oh, I see, it's actually not in a section — that might be worth stating explicitly!

**Comment by glasser** [comment]
Actually experimenting with this for once, it looks like the impact of this (similar to, say, a TOML parse error) is that mise does not update PATH (etc) but also doesn't actively clear PATH entries from previous mise runs.

**Comment by erickguan** [comment]
> it's actually not in a section — that might be worth stating explicitly!

I would keep this out of this PR because other sections follow similar style.

> Actually experimenting with this for once, it looks like the impact of this (similar to, say, a TOML parse error) is that mise does not update PATH (etc) but also doesn't actively clear PATH entries from previous mise runs.

Good find. This looks like a problem but a trivial one. It's easy to reload the interactive shell for mise users. If there are more complains, we could look into it. What do you think?

File: docs/configuration.md [hunk 1]
```diff
@@ -2,23 +2,23 @@
 
 ## `mise.toml`
 
-`mise.toml` is the config file for mise. They can be at any of the following file paths (in order of precedence, top overrides configuration of lower paths):
```
Comment (jdx) [inline docs/configuration.md]: change these back

File: docs/configuration.md [hunk 2]
```diff
@@ -130,7 +130,13 @@ my_custom_node = '20'
...
 ### Minimum mise version
 
-Define a minimum supported version of mise for the config file. mise will ignore config files that use too new of a version.
+Specify the minimum supported version of mise required for the configuration file.
+If the configuration file specifies a version of mise that is higher than
+the currently installed version, mise will:
+
+- Ignore the configuration file.
+- Output a warning message.
+- Terminate execution.
```
Comment (jdx) [inline docs/configuration.md]:
```suggestion
the currently installed version, mise will error out.
```

**Pull request merged**